hazelcast/src/main/java/com/hazelcast/query/impl/Metadata.java

```diff
@@ -21,11 +21,16 @@
  * is object. It is up to the user of this class to determine the type of
  * metadata and act accordingly.
  */
-public class Metadata {
+public class Metadata implements JsonMetadata {
```

Comment:
sorry, i'm caring with the naming because we have many similar metadata classes for different purposes, trying to increase readability with clear naming. seemed this can be:
```suggestion
public class Metadata implements JsonMetadataImpl {
```

Reply:
`JsonMetadata` is an interface. I think JsonMetadataImpl is very misleading in this context.

Reply:
it was a typo, updated my comment